tofu/tests/tests09_tutorials/tests01_runall.py

```diff
@@ -17,7 +17,7 @@
 
 
 _HERE = os.path.abspath(os.path.dirname(__file__))
-_TFROOT = _HERE[:-_HERE[::-1].index('/tofu'[::-1])-len('/tofu')]
+_TFROOT = os.path.abspath(os.path.join(os.path.dirname(_HERE), os.pardir))
```

Comment:
Smart move cowboy ! ;-)
But... although the use of os.pardir is a good idea, the result is not equivalent to the original.

Indeed, for example on my platform (where I have several nested /tofu directories):
```
In [7]: _HERE
Out[7]: '/Home/DV226270/ToFu_All/tofu_git/tofu/tofu/tests/tests09_tutorials'

In [8]: _TFROOT = _HERE[:-_HERE[::-1].index('/tofu'[::-1])-len('/tofu')]

In [9]: _TFROOT2 = os.path.abspath(os.path.join(os.path.dirname(_HERE), os.pardir))

In [10]: _TFROOT
Out[10]: '/Home/DV226270/ToFu_All/tofu_git/tofu'

In [11]: _TFROOT2
Out[11]: '/Home/DV226270/ToFu_All/tofu_git/tofu/tofu'
```

Here by _TFROOT, we mean the directory from which you should be able to import the local tofu (i.e. from which tofu is seen as a package), which one step upwards from your solution (in your case you're already inside the package itself, so you can't import it).

Reply:
Okay, I think I can update this one to get the desired behaviour.